### PR TITLE
Modify build and deploy scripts

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Seperate build and deploy scripts.

Also rename and enable running as bash
script.

i.e.
old:
```
$ builtit bash dev
```

new:
```
$ ./k8sbuild.sh
$ ./k8sdeploy.sh dev
```

#### Why
Seprate scripts so that you can, potentially, change deployments
(uncommitted) and deploy them without committing. Such as for
adding env vars and deploying the same build to multiple environments.

Could extend the deploy to take a shah to deploy an existing build but not the
current.

### CODE REVIEW

File: k8sdeploy.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  version=$(git rev-parse $(git branch | grep \* | cut -d ' ' -f2))
+  docker_registry='754256621582.dkr.ecr.eu-west-2.amazonaws.com/laa-get-paid/cccd'
+  docker_tag=app-${version}
+  docker_image=${docker_registry}:${docker_tag}
+
+  echo "Deploying $docker_image to $environment on cluster $context"
+  kubectl set image -f kubectl_deploy/${environment}/deployment.yaml cccd-app=$docker_image --local -o yaml | kubectl apply -n cccd-${environment} --context ${context} -f -
+}
+
+_k8sdeploy $@
```
Comment (jsugarman) [inline k8sdeploy.sh]: line break

File: k8sdeploy.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (jsugarman) [inline k8sdeploy.sh]: change to sh

File: k8sbuild.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+function _k8sbuild() {
+  usage="k8build -- build, tag and push image to ecr"
+
+  REGION='eu-west-2'
```
Comment (jsugarman) [inline k8sbuild.sh]: lower case for local/temp vars
https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization

File: k8sbuild.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  CONTEXT='live-1'
```
Comment (jsugarman) [inline k8sbuild.sh]: not used

File: k8sbuild.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  echo '------------------'
+  echo $docker_image
+  echo '------------------'
+
+  $(aws ecr --profile "$AWS_PROFILE" get-login --no-include-email --region "$REGION")
+  docker build \
+          --build-arg VERSION_NUMBER=$docker_image \
+          --build-arg BUILD_DATE=$(date +%Y-%m-%dT%H:%M:%S%z) \
+          --build-arg COMMIT_ID=$VERSION \
+          --build-arg BUILD_TAG=$CCCD_BUILD_TAG \
```
Comment (jsugarman) [inline k8sbuild.sh]: should be docker_tag

File: k8sdeploy.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+function _k8sdeploy() {
+  usage="k8sdeploy -- build, tag and push image to ecr
```
Comment (colinbruce) [inline k8sdeploy.sh]: `build, tag and deploy to k8s environment`?

**Review by colinbruce — APPROVED** [review]
Generally fine, one minor tweak to the description.

Also, is the `api-sandbox` deploy file supposed to be in here? It's not mentioned in the PR description.

👍

File: k8sdeploy.sh (hunk 6 above)
Comment (jsugarman) [reply k8sdeploy.sh]: copy paste 💥 - cheers

**Pull request merged**